File: 2913-subarrays-distinct-element-sum-of-squares-i/2913-subarrays-distinct-element-sum-of-squares-i.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var sumCounts = function(nums) {
    let i=0,j=0;
    let sum=0;
    while(j<nums.length){
        let s=new Set(nums.slice(i,j+1));
        sum+=s.size*s.size;
        if(j==nums.length-1){
            i++;
            j=i;
            continue;
        }
        j++;
    }
    return sum;
};

File: 2913-subarrays-distinct-element-sum-of-squares-i/2913-subarrays-distinct-element-sum-of-squares-i.cpp
class Solution {
public:
    int sumCounts(vector<int>& nums) {
        int i=0,j=0;
        int sum=0;
        while(j<nums.size()){
            set<int>s(nums.begin()+i,nums.begin()+j+1);
            sum+=s.size()*s.size();
            if(j==nums.size()-1){
                i++;
                j=i;
                continue;
            }
            j++;
        }
        return sum;
    }
};